### PR TITLE
Start Task_RingsMgrMain

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: src/game/game.c [hunk 1]
```diff
@@ -164,8 +162,8 @@ void CreateGameStage(void)
     gUnknown_03005424 |= 0x21;
     gUnknown_03005424 &= ~0x80;
 
-    gUnknown_03005430 = 0;
-    gUnknown_0300542C = 10;
+    gBossRingsShallRespawn = FALSE;
+    gBossRingsRespawnCount = BOSS_RINGS_DEFAULT_RESPAWN_COUNT;
```
Comment (freshollie) [inline src/game/game.c:166]: oooo

File: src/game/interactables_2/egg_utopia/cannon.c [hunk 2]
```diff
@@ -257,8 +261,10 @@ NONMATCH("asm/non_matching/sub_807E66C.inc",
     if (PLAYER_IS_ALIVE) {
         // Maybe log
         {
+#ifndef NON_MATCHING
```
Comment (freshollie) [inline src/game/interactables_2/egg_utopia/cannon.c:264]: Fair enough, but cos this doesn't yet match at all I didn't add them

**Review by freshollie — APPROVED** [review]

File: src/game/game.c (hunk 1 above)
Comment (JaceCear) [reply src/game/game.c:166]: This is the amount of times *all* rings spawn btw. (in a boss stage)
Couldn't think of a better name!

Comment (JaceCear) [reply src/game/game.c:166]: And I'm not 100% sure whether it's limited to bosses, since it seems to be used in the Ring-Collect mode

**Pull request merged**